### PR TITLE
1699 - Implement Compact Form

### LINKED ISSUE #1699: Form-compact: New form-compact initial foundation
**Is your feature request related to a problem? Please describe.**
This is the foundation for the form-group visual updates.

**Describe the solution you'd like**
Build the foundation for the new form-compact component per the design skeleton attached. The code should be production worthy and be considered the modular base infrastructure for future enhancements (like dropdowns, masking, dirty states, searches, lookups - some of which can be referenced in infor-design/design-system#248).

Please feel free to reach out to anyone with questions and leave comments on this issue as necessary.

Architecture:
- [x] Build it as a new component `form-compact`
    - sass, js, readme.md, etc
- [ ] Make sure the component works with the custom builder
- [x] Provide a full width `example-index.html` demo page
- [x] Add basic documentation to `/readme.md` as necessary
- [x] Ignore writing any tests for now

Code/Development:
- [x] Be sure to use a 12-col grid without gutters for responsiveness
- [x] Be conscious of focus state & tabIndex
- [x] Note the ellipsis for test and labels
- [x] All section headers should have the gray background to match "section three" 
    - Note this is *not* represented in the design. That's okay.

**Describe alternatives you've considered**
n/a

**Additional context**
mock compact v4 white bg

### PR DESCRIPTION
**Pull request by EdwardCoyle**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR implements a forms style for IDS for field-heavy forms.

**Related github/jira issue (required)**:
Closes #1699 

**Steps necessary to review your pull request (required)**:
- Pull this branch and run the demoapp
- See the examples in http://localhost:4000/components/form-compact/list

_NOTE:_ This PR needs a merge from `master` once #1728 is merged in order to properly test

### CODE REVIEW

**Comment by deep7102** [comment]
Looking Good! only one thing saw, I think its just the example

http://localhost:4000/components/form-compact/example-list-detail.html
- Open in mobile view
- Click on any item to go and show the form
- See here cannot scroll to all the way down to see "Section Three" stuff

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 ### v4.17.0 Features
 
 - `[Datagrid]` Added support to cancel `rowactivated` event. Now it will trigger the new event `beforerowactivated` which will wait/sync to cancel or proceed to do `rowactivated` event. ([#1021](https://github.com/infor-design/enterprise/issues/1021))
+- `[Form Compact]` Implemented design for field-heavy forms. ([#1699](https://github.com/infor-design/enterprise/issues/1699))
```
Comment (nickwynja) [inline docs/CHANGELOG.md]: I think you should mention that this design is still "experimental", likely not production ready, and subject to change without notice.

**Comment by nickwynja** [comment]
>go to mobile size and you see Section One then Section Two (the headings are out of order)

@EdwardCoyle I think we can just remove this. @tmcconechy This example was the original design that we moved away from because of the exact layout issue you noticed.

>What about dropdowns and other field types? Is that "later?

Yes. We want to get the fundamental style and layouts right first through a series of user testing before extending the functionality.

**Comment by nickwynja** [comment]
@tmcconechy Yes, agreed.

**Pull request merged**